### ISSUE TITLE
Serial Port CTS emulation switch

### ISSUE THREAD

**Issue by Koatman** [joncampbell123/dosbox-x#462]
Hi!
'I am trying to program an old Vertex radio.  The software expects a clear to send signal to be present on the serial port.  I am fairly sure this is not being passed from linux host into dosbox.  As a result, the emulator software doesn't think the cable is plugged in, and doesn't work.  I can read and even write back the read file from linux with jplevulator.  The cable and radio are fine.

If there were a way to pass this CTS signal into the emulator by pressing a button or clicking a check-box, I think this would fix this problem.

K Chapman

**Comment by joncampbell123** [comment]
Is this through a USB to serial adapter?

According to this pinout, the 9-pin form most adapters take has a CTS pin.

http://www.usconverters.com/index.php?main_page=page&id=61

So perhaps the bug is that the serial passthrough is not passing through CTS.

It's also possible from experience that the serial cable may omit it or tie it to another line, or the USB serial adapter doesn't provide it.

**Comment by Koatman** [comment]
Jon,  This is a real serial port. I have tried this on two different computers.  A couple of times, I was able to get the program to continue, when the port / hardware was crashed, as I had miswired the adapter cable I built from scratch.  The error was type -7, if I recall correctly.  Restarting the computer cleared this problem. # **This indicates that the software is looking for a specific state of the serial port.**

I'm not sure what the exact issue here is, but there is something about the serial port which is not being emulated properly or cannot be emulated.  Does L:inux pass the CTS pin state through to the ttyS0? I think that it a plain character device.  I know that DOS can manipulate the serial port directly on bare metal.
